### PR TITLE
Use wrappers for boolean types

### LINKED ISSUE #14: Not able to check if a primitive type field (bool, string, int) has been set or not
I am consuming the proto in java agent and I want to provide default values for header/body capture (enabled by default) but also reporting address and secure reporting. 

The reporting address is easy to set. If it is empty I will just set it to a default value. However the problem is with `bool` values e.g. `reporting.secure` and `dataCapture.httpHeaders.request` etc. These values I am not able to check if they have been set by the user or not. Specifically, I am not able to determine if the user explicitly set the `bool` to false.

```java
FieldDescriptor requestField = httpBodyBuilder.getDescriptorForType().findFieldByName("request");
    if (!httpBodyBuilder.hasField(requestField)) { // returns always false even if the field has been set to false.
      httpBodyBuilder.setRequest(true);
    }
```

Here is how I create the config. 
```java
  public static AgentConfig load(String filename) throws IOException {
    File configFile = new File(filename);
    InputStream fileInputStream = new FileInputStream(configFile);
    String json = convertYamlToJson(fileInputStream);

    AgentConfig.Builder configProtoBuilder = AgentConfig.newBuilder(); // if the applyDefaults is called here the PARSER fails on com.google.protobuf.InvalidProtocolBufferException: Field AgentConfig.serviceName has already been set.
    PARSER.ignoringUnknownFields().merge(json, configProtoBuilder);
    AgentConfig agentConfig = configProtoBuilder.build();

    return applyDefaults(agentConfig.toBuilder()).build();
  
```

see also https://github.com/protocolbuffers/protobuf/issues/359

I propose to use `optional` in the filed, however it is an experimental feature or use wrappers which is a recommended approach - https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wrappers.proto and https://developers.google.com/protocol-buffers/docs/reference/google.protobuf
``

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <pavol.loffay@traceable.ai>

Use wrappers to be able to check if the value was explicitly set to false by user.

Resolves #14

### CODE REVIEW

**Review by mohit-a21 — APPROVED** [review]

File: config.proto [hunk 1]
```diff
@@ -21,16 +23,16 @@ message Reporting {
...
 
 // Message describes what message should be considered for certain DataCapture option
 message Message {
     // request describes the outgoing/incoming message in a client/request operation
-    bool request = 1;
+    google.protobuf.BoolValue request = 1;
 
     // response describes the incoming/outgoing message in a client/request operation
-    bool response = 2;
+    google.protobuf.BoolValue response = 2;
```
Comment (mohit-a21) [inline config.proto:35]: we should use the wrappers for all primitives.

Comment (pavolloffay) [reply config.proto:35]: agree I have also used it for string.

**Comment by jcchavezs** [comment]
While I like this change I think it will break the config in the GoAgent (this is the kind of problems that come up when we have that logic in another repo) so I will check and let you know.

**Comment by pavolloffay** [comment]
nit: if you are using git submodules try `g fetch up pull/15/head:PR15`  and then checkout PR15 branch in the directory with the config.

In Golang you should be able to use the wrappers and check if the value has been set or not. Maybe we could also remove the custom generator code.

**Comment by jcchavezs** [comment]
Let's make it happen.

**Comment by jcchavezs** [comment]
@pavolloffay could you please add as convention that scalar values should be wrapped and explain the why?

**Comment by pavolloffay** [comment]
yeah I can document that

**Pull request merged**